kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemView.java

```diff
@@ -214,6 +215,24 @@ public long getLastModified() {
         throw new DatasetIOException("Cannot find last modified time of of " + dir, e);
       }
     }
+
+    // if view was marked ready more recently count it as the modified time
+    long readyTimestamp = ((FileSystemDataset<E>)dataset).getReadyTimestamp(getConstraints());
+    if(lastMod < readyTimestamp) {
+      lastMod = readyTimestamp;
+    }
+
     return lastMod;
   }
+
+  @Override
+  public void signalReady() {
+    ((FileSystemDataset<E>)dataset).signalReady(getConstraints());
```

Comment:
I don't think we should cast to `FileSystemDataset` here. I'd prefer to either pass the signal manager to the views, like we do with the partition listener, or find some other way, like adding getSignalManager to AbstractRefinableView.